productivity-suite/app/fragments/news/src/routes/[...news].tsx

```diff
@@ -16,10 +17,13 @@ export default function News() {
 
   const [page, setPage] = createSignal<number>(1);
 
-  getBus(ref).listen<{ page: number }>("news-page", ({ page }) => {
-    if (isFinite(page) && page > 0) {
-      setPage(page);
-    }
+  onMount(() => {
+    getBus(ref).listen<{ page: number }>("news-page", ({ page }) => {
+      if (isFinite(page) && page > 0) {
+        setPage(page);
+      }
+    });
+    getBus(ref).onCleanup(() => getBus(ref).dispatch("change-news-page", {}));
```

Comment:
Here we add a handler that will be called when the fragment itself is being removed.